eng/tools/generator/cmd/v2/common/workspaceProcessor.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package common
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/eng/tools/generator/repo"
+)
+
+var (
+	commitIDRegex = regexp.MustCompile("^[0-9a-f]{40}$")
+)
+
+func GetSDKRepo(sdkRepoParam, sdkRepoURL string) (*repo.SDKRepository, error) {
```

Comment:
If I did not remember wrong, `repo.SDKRepository` is an interface, therefore we did not need to return pointer here

Reply:
Changed.